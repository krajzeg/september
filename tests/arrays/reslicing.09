array := [0,1,2,3,4,5]
double_slice := array[1..4][1..2]

print("After slicing twice, the result should be 2,3:", double_slice)

called := 0
Slice:::_at = Slice.at
Slice:::at = |index| {
	called = called + 1
	this._at(index)
}

print("If the double slice works correctly, this will be 3:", double_slice[1])
print("Slice.at is called", called, "times on each access (should be once).")
