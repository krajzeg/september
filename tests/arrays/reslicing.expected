After slicing twice, the result should be 2,3: 2, 3
If the double slice works correctly, this will be 3: 3
Slice.at is called 1 times on each access (should be once).
